beacon_node/rest_api/src/validator.rs

```diff
@@ -147,3 +134,189 @@ pub fn get_validator_duties<T: BeaconChainTypes + 'static>(req: Request<Body>) -
     );
     Ok(success_response(body))
 }
+
+/// HTTP Handler to produce a new BeaconBlock from the current state, ready to be signed by a validator.
+pub fn get_new_beacon_block<T: BeaconChainTypes + 'static>(req: Request<Body>) -> ApiResult {
+    let beacon_chain = get_beacon_chain_from_request::<T>(&req)?;
+
+    let query = UrlQuery::from_request(&req)?;
+    let slot = match query.first_of(&["slot"]) {
+        Ok((_, v)) => Slot::new(v.parse::<u64>().map_err(|e| {
+            ApiError::InvalidQueryParams(format!("Invalid slot parameter, must be a u64. {:?}", e))
+        })?),
+        Err(e) => {
+            return Err(e);
+        }
+    };
+    let randao_reveal = match query.first_of(&["randao_reveal"]) {
+        Ok((_, v)) => Signature::from_bytes(
+            hex::decode(&v)
+                .map_err(|e| {
+                    ApiError::InvalidQueryParams(format!(
+                        "Invalid hex string for randao_reveal: {:?}",
+                        e
+                    ))
+                })?
+                .as_slice(),
+        )
+        .map_err(|e| {
+            ApiError::InvalidQueryParams(format!("randao_reveal is not a valid signature: {:?}", e))
+        })?,
+        Err(e) => {
+            return Err(e);
+        }
+    };
+
+    let new_block = match beacon_chain.produce_block(randao_reveal, slot) {
```

Comment:
This could be

```rust
let new_block = beacon_chain.produce_block(randao_reveal, slot)
  .map(|(block, _state)| block)
  .map_err(|e| ApiError::ServerError(format!("Beacon node is not able to produce a block: {:?}", e)))))?
```

Note: I haven't tested this, it probably doesn't compile.